sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFnAdapters.java

```diff
@@ -86,6 +86,7 @@ public void setup() throws Exception {
 
     @Override
     public void startBundle(Context c) throws Exception {
+      fn.prepareForProcessing();
```

Comment:
I think we had the same discussion in previous PR -- I'm worried about moving the call to `prepareForProcessnig` into the `DoFnAdapters`, since the `DoFnAdapters` will be going away (which is why they're marked `@Deprecated`).

Why does this PR need to move the call out from where it was? If anything, it should be easier to keep it in the `beforeDelegation` section of the `StartBundleMethodDelegation`.